docs/matchers/index.rst

```diff
@@ -0,0 +1,388 @@
+Matchers
+==========
+
+Matchers provide a way to validate the arguments passed to :doc:`../mock_callable/index`, :doc:`../mock_async_callable/index` and :doc:`../mock_constructor/index`
+
+Basic Usage
+-----------------------
+
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+  class TestBackupDelete(testslide.TestCase):
+    def setUp(self):
+      super().setUp()
+      self.storage_mock = testslide.StrictMock(storage.Client)
+      # Makes storage.Client(timeout=60) return self.storage_mock
+      self.mock_constructor(storage, 'Client')\
+        .for_call(timeout=matchers.ThisInt(60))\
```

Comment:
I'd consider putting a few words before, explaining what's going on here, like "let's say you want to accept any timeout value, as long as it is an integer".